### PR TITLE
remove default value as parameter in ParseHost()

### PR DESCRIPTION
**Pull request by azurezk**
Signed-off-by: Kun Zhang zkazure@gmail.com

remove default value as parameter in /pkg/parsers.ParseHost()

### CODE REVIEW

File: pkg/parsers/parsers.go [hunk 1]
```diff
@@ -12,17 +12,29 @@ import (
 	"strings"
 )
 
+var (
+	// DefaultHTTPHost Default HTTP Host used if only port is provided to -H flag e.g. docker -d -H tcp://:8080
+	DefaultHTTPHost = "127.0.0.1"
```
Comment (runcom) [inline pkg/parsers/parsers.go]: I think these vars don't belong to this pkg cause they are docker specific

Comment (azurezk) [inline pkg/parsers/parsers.go]: How about move ParseHost() and ParseTCPAddr(), ParseUnixAddr() it relies on to opt, cause its only used in opt?  @runcom

**Comment by runcom** [comment]
@azurezk I think only the FIXME you are trying to fix is to be removed.

I think it's good to have a pkg which parses host/addrs and the like and have methods receiving default args if things go wrong. To me it's good design if a pkg  it's not coupled with anything, you just pass a default value and that's it and it's reusable (and testable).
This is my point of view, I'll leave to others to comment on this cause maybe they don't agree

File: pkg/parsers/parsers.go (hunk 1 above)
Comment (icecrime) [inline pkg/parsers/parsers.go]: All things in `pkg/` are meant to be usable by other projects. Here, `pkg/parsers` has a purpose of its own, so I think it's a sane thing to keep it.

**Comment by cpuguy83** [comment]
@azurezk Thanks for looking at this `FIXME` in the code, we definitely have lots of these!
Because this isn't really the right fix as @runcom stated I'm going to close.
I think we can probably remove this particular `FIXME`

Thanks again!

**Pull request closed without merging**